Src/GoogleApis.Auth/OAuth2/Requests/GoogleAuthorizationCodeRequestUrl.cs

```diff
@@ -58,6 +59,13 @@ public class GoogleAuthorizationCodeRequestUrl : AuthorizationCodeRequestUrl
         [Google.Apis.Util.RequestParameterAttribute("include_granted_scopes",
             Google.Apis.Util.RequestParameterType.Query)]
         public string IncludeGrantedScopes { get; set; }
+        
+        /// Gets or sets a collection of user defined query parameters to facilitate any not explicitly supported
+        /// by the library which will be included in the resultant authentication URL.
+        /// </summary>
+        [Google.Apis.Util.RequestParameterAttribute("user_defined_query_parameters",

```

Comment:
1. I'm still not sure it's the right place to have it, I'll keep reviewing the code and update this comment.
2. I'm pretty sure that we don't need Google.Apis.Util.RequestParameterAttribute here, since it's not field recognized by the server. We only add this for field that the server knows, so we can have something like "user_defined_query_patameters=blahblahblah in the URL